src/NServiceBus.Persistence.AzureStorage/Outbox/OutboxRecord.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace NServiceBus.Persistence.AzureStorage
+{
+    using System;
+    using System.Collections.Generic;
+    using Outbox;
+    using Newtonsoft.Json;
+    using Microsoft.Azure.Cosmos.Table;
+
+    class OutboxRecord : TableEntity
+    {
+        [IgnoreProperty]
+        public string Id
+        {
+            get => RowKey;
+            set => RowKey = value;
+        }
+
+        public bool Dispatched { get; set; }
+
+        public string TransportOperations { get; set; }
+
+        public string DispatchedAt { get; set; }
+
+        [IgnoreProperty]
+        public TransportOperation[] Operations { get; set; } = Array.Empty<TransportOperation>();
```

Comment:
Potentially describing the trick with ignoring transport operations and writing those as a string on serialization and back when deserializing will be good.